cpp/models/ide_secir/model.cpp

```diff
@@ -60,12 +60,17 @@ void Model::compute_compartment_from_flows(ScalarType dt, Eigen::Index idx_Infec
                                            Eigen::Index idx_IncomingFlow, int idx_TransitionDistribution1,
                                            int idx_TransitionDistribution2)
 {
-    ScalarType sum = 0;
-
+    ScalarType sum       = 0;
+    ScalarType calc_time = 0;
     // Determine relevant calculation area and corresponding index.
-    ScalarType calc_time =
-        std::max(parameters.get<TransitionDistributions>()[idx_TransitionDistribution1].get_support_max(dt, m_tol),
-                 parameters.get<TransitionDistributions>()[idx_TransitionDistribution2].get_support_max(dt, m_tol));
+    if ((1 - parameters.get<TransitionProbabilities>()[idx_TransitionDistribution1]) > 0) {
```

Comment:
```suggestion
    if ((parameters.get<TransitionProbabilities>()[idx_TransitionDistribution2]) > 0) {
```
I find that a bit more intuitive but don't have a strong opinion here.

Reply:
That is actually wrong because idx_TransitionDistribution2 can be a random index if the second distribution is not used (eg for E->C). This is the whole problem of this issue/PR. 

I used 1- prob[idx]>0 because this is the factor used in the calculation below, another opportunity would of course be prob[idx1]<1.

Reply:
But if the second TransitionDistribution is not used the default index is 0 and for this the probability is 1. But maybe I'm confused here.

But it does make sense to use the first Transition Distribution in the if statement because this already determines if we need both distributions or not, so it is fine as it is.

Reply:
Yes you are right, we actually have a test that ensures that this probability is one. :) I still prefer the other solution.